web_search_with_and/static/src/js/search.js

```diff
@@ -0,0 +1,53 @@
+odoo.define('web_search_with_and', function (require) {
+    "use strict";
+
+    var SearchView = require('web.SearchView');
+    var Backbone = window.Backbone;
+
+    SearchView.include({
+        // Override the base method to detect a 'shift' event
+        select_completion: function (e, ui) {
+            if (e.shiftKey
+                && ui.item.facet.values
+                && ui.item.facet.values.length
+                && String(ui.item.facet.values[0].value).trim() !== "") {
+                // In case of an 'AND' search a new facet is added regarding of the previous facets
+                e.preventDefault();
+
+                this.query.add(ui.item.facet, {shiftKey: true});
+            } else {
+
+                return this._super.apply(this, arguments);
+            }
+        }
+    });
+
+    SearchView.SearchQuery.prototype = SearchView.SearchQuery.extend({
```

Comment:
Alright sounds like a good enough reason for me, thanks for the elaboration @Creamaster.

Because this is non-standard, can you add a docblock indicating the why so that future module maintainers don't have to solve the same problem repeatedly?